### DISCUSSION TITLE
A Flatpak version for other Linux distros

### DISCUSSION THREAD

**Discussion by archerallstars** [o3de/o3de#9289] [Ideas]
Currently, the Linux binary to download and install O3DE is only available to the Debian-based Linux distros. I think it's a must to provide the engine to other Linux distros as well. For instance, Fedora users whether through .rpm or Flatpak.

**Comment by Binghoul** [comment]
Arch Linux  (K.I.S.S.)  
Rolling updates  : )

**Reply by L4exp** [reply]
1) In Arch - updates adds with some delay
2) We have solution for deb and arch package system. E.g. calculate linux does't support direct installation.

Ofcourse there is way to unpack deb package and do some script after that(as required).

P.S.(from arch packages page)
https://aur.archlinux.org/packages/o3de-nightly-bin

First Submitted:	2022-02-17 21:43 (UTC)
Last Updated:	        2022-04-21 02:53 (UTC)

**Reply by Binghoul** [reply]
Thanks, ill try again when 22.05 comes out.
Just curious how far behind is the linux development (if any)  from Windows? features and such...

**Reply by L4exp** [reply]
Technically feature set is same or close to same on both platforms.
Simplified - important steps to build engine:
1) Downloading python and installing deps(python - crossplatform)
2) Source code compiling by cmake(cmake - crossplatform)

But not all features works correctly.
There is a page which contains issues related with linux platform and this is important.
They described:
https://github.com/o3de/o3de/projects/13

**Comment by matteogrs** [comment]
As a possible alternative for other distros, I've create an experimental tool that builds and runs the engine inside constrained containers. The result is similar to a Flatpak application, since there is no need to install any dependency on the main operating system: https://github.com/loherangrin/o3tanks

At the moment the tool requires to build the engine from its source code (it is done automatically), since it allows more scenarios than the pre-built package (e.g. it can create `debug` and  `release` profiles. The latter is required to export games).

The tool also supports to install the engine on the main OS directly, for who doesn't want to use containers. It suggests which system packages are missing and then invokes cmake commands. Now only the following OS are checked: `arch`, `debian`, `fedora` and `opensuse leap`.
